MergerLogic/Utils/PathPatternUtils.cs

```diff
@@ -0,0 +1,92 @@
+﻿using MergerLogic.DataTypes;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MergerLogic.Utils
+{
+    public class PathPatternUtils
+    {
+        private string[] parts;
+        private string[] keys;
+        private Dictionary<string, string> keyValues;
+
+        public PathPatternUtils(string pattern)
+        {
+            this.parts = new string[4];
+            this.keys = new string[3];
+            this.keyValues = new Dictionary<string, string>(9);
+            this.compilePattern(pattern);
+        }
+
+        private void compilePattern(string pattern)
```

Comment:
Why not just split by {x} and so on depending on the source type?

Reply:
this is more generic then multiple splits (order of Parma's don't matter) and should be faster too

Reply:
i want to see benchmark tests for:
String.Format
StringBuilder
String.replace
Regex
Current existing code in PR

For Build and Render